src/Repository/DefaultUnleashProxyRepository.php

```diff
@@ -0,0 +1,207 @@
+<?php
+
+namespace Unleash\Client\Repository;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestFactoryInterface;
+use Symfony\Component\VarExporter\Exception\LogicException;
+use Unleash\Client\Configuration\Context;
+use Unleash\Client\Configuration\UnleashConfiguration;
+use Unleash\Client\Configuration\UnleashContext;
+use Unleash\Client\DTO\DefaultProxyFeature;
+use Unleash\Client\DTO\Feature;
+use Unleash\Client\DTO\ProxyFeature;
+
+final class DefaultUnleashProxyRepository implements ProxyRepository
+{
+    public function __construct(
+        private UnleashConfiguration $configuration,
+        private ClientInterface $httpClient,
+        private RequestFactoryInterface $requestFactory
+    ) {
+    }
+
+    public function findFeatureByContext(string $featureName, ?Context $context = null): ?ProxyFeature
+    {
+        $cacheKey = $this->getCacheKey($featureName, $context);
+
+        if ($this->configuration->getCache() !== null && $this->configuration->getCache()->has($cacheKey)) {
+            $cachedFeature = $this->configuration->getCache()->get($cacheKey);
+            if (is_array($cachedFeature)) {
+                $featureData = $this->validateResponse($cachedFeature);
+                if ($featureData !== null) {
+                    return new DefaultProxyFeature($featureData);
+                }
+            }
+        }
+
+        $context ??= new UnleashContext();
+        $featureUrl = $this->configuration->getUrl() . 'frontend/features/' . $featureName;
```

Comment:
Just because I don't know the answer: Do we need to url-encode `$featureName`? I think this is the first time we add a feature name in a URL: https://sourcegraph.com/search?q=context%3Aglobal+featureName+repo%3A%5Egithub%5C.com%2FUnleash%2Funleash-client-php%24+&patternType=standard&sm=1&groupBy=path

Reply:
It is the first time, yea. I don't think we need to url encode this. You shouldn't ever be targeting a non url safe feature name because you can't get one into unleash itself. If you misspell it, this won't work anyway even with a nicely encoded name